<?xml version="1.0"?>
<launch>
	<param name="/use_sim_time" value="true" />

	<arg name="init_x" default ="0" />
  	<arg name="init_y" default ="0.35" />
  	<arg name="init_a" default ="0" />

	 <!-- start gazebo with the simulated scenario -->
    <include file="$(find robotics_challenge)/launch/turtlebot_world.launch">
		<arg name="goal_x" value ="1.13" />
		<arg name="goal_y" value ="1.92" />
		<arg name="world_file" value="$(find robotics_challenge)/worlds/challenge5.world" />
		<arg name="init_x" value ="$(arg init_x)" />
    	<arg name="init_y" value ="$(arg init_y)" />
    	<arg name="init_a" value ="$(arg init_a)" />
	</include>

	<!--Start the map server and the costmap for the path planner-->
 	<node name="map_server" pkg="map_server" type="map_server" args="$(find robotics_challenge)/maps/house_reduced.yaml"/>
	 <node pkg="costmap_2d" type="costmap_2d_node" name="costmap_2d" output="screen">
        <rosparam command="load" file="$(find robotics_challenge)/cfg/global_costmap_params.yaml" />
    </node>

	<!--Start the Montecarlo Localization module-->
	<node pkg="amcl" type="amcl" name="amcl" args="scan:=/scan">
		<param name="initial_pose_x" value="$(arg init_x)" /> 
    	<param name="initial_pose_y" value="$(arg init_y)" />
		<param name="initial_pose_a" value="$(arg init_a)" />
	</node>
	
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find robotics_challenge)/rviz/turtlebot.rviz"/>
</launch>
